### PR TITLE
Update to .NET 7-era packages

### PR DESCRIPTION
**Pull request by rainersigwald**
Update references to packages published from the `dotnet/runtime` repo. This must be done in coordination with the VS repo, which is pending at (Microsoft-internal link) https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/435695.

Test insertion from the experimental branch that showed passing VS perf tests (internal again): https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/446441.

### CODE REVIEW

**Review by MichalPavlik — APPROVED** [review]
LGTM

File: eng/Versions.props [hunk 1]
```diff
@@ -26,21 +26,21 @@
...
         and follow the guidelines written here (internal-link): https://dev.azure.com/devdiv/DevDiv/_wiki/wikis/DevDiv.wiki/1796/How-to-add-a-Known-Issue
     -->
     <SystemMemoryVersion>4.5.5</SystemMemoryVersion>
     <SystemNetHttpVersion>4.3.4</SystemNetHttpVersion>
-    <SystemReflectionMetadataLoadContextVersion>6.0.0</SystemReflectionMetadataLoadContextVersion>
-    <SystemReflectionMetadataVersion>6.0.0</SystemReflectionMetadataVersion>
-    <SystemResourcesExtensionsPackageVersion>6.0.0</SystemResourcesExtensionsPackageVersion>
-    <SystemSecurityPermissionsVersion>6.0.0</SystemSecurityPermissionsVersion>
+    <SystemReflectionMetadataLoadContextVersion>7.0.0</SystemReflectionMetadataLoadContextVersion>
+    <SystemReflectionMetadataVersion>7.0.0</SystemReflectionMetadataVersion>
```
Comment (am11) [inline eng/Versions.props:39]: @ViktorHofer, @jkoritzinsky, @trylek re: https://github.com/dotnet/arcade/issues/10516

Should we also update this in host model? https://github.com/dotnet/runtime/blob/1f8d401995fe904c69d28a05bef7948d6affe2c2/src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj#L22-L23 @mmitche has updated it in the SDK repo.

This came up yesterday in another discussion https://github.com/dotnet/runtime/pull/85038#discussion_r1174593791. I think we can unify SRM version to 7.0 across the runtime repo (provided Tomas' ILSpy concern is addressed).

Comment (ViktorHofer) [reply eng/Versions.props:39]: Could we re-target HostModel to net7.0/net8.0? That would solve the PackageReference issue and would remove the prebuild on S.T.J and S.R.M.

Comment (jkoritzinsky) [reply eng/Versions.props:39]: Yeah Microsoft.NET.HostModel is used by the sdk to implement the tasks that manipulate the apphosts. The dependency requirements for that package are that the dependencies are compatible with being loaded by an MSBuild task in dotnet/sdk build/publish scenarios. We can make any changes that are compatible with that rule (including package upgrades/downgrades, etc).

**Pull request merged**